tests/LSP6KeyManager/tests/LSP6ControlledToken.test.ts

```diff
@@ -0,0 +1,840 @@
+        );
+        const payload = context.token.interface.encodeFunctionData(
+          "setData(bytes32,bytes)",
+          [key, value]
+        );
+
+        await context.keyManager.connect(context.accounts[4]).execute(payload);
+
+        expect(await context.token["getData(bytes32)"](key)).to.be.equal(value);
+      });
+
+      it("should restrict second controller with AllowedERC725YKeys", async () => {
+        const key =
+          ERC725YKeys.LSP6["AddressPermissions:AllowedERC725YKeys"] +
+          context.accounts[4].address.substring(2);
+        const value = abiCoder.encode(
+          ["bytes32[]"],
+          [
+            [
+              ethers.utils
+                .keccak256(ethers.utils.toUtf8Bytes("FirstRandomString"))
+                .substring(0, 34) + ARRAY_INDEX.ZERO,
+            ],
+          ]
+        );
+        const payload = context.token.interface.encodeFunctionData(
+          "setData(bytes32,bytes)",
+          [key, value]
+        );
+
+        await context.keyManager.connect(context.accounts[0]).execute(payload);
+
+        expect(await context.token["getData(bytes32)"](key)).to.be.equal(value);
+      });
+
+      it("should change allowed keys", async () => {
+        const keys = [
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("FirstRandomString")),
+          ethers.utils
+            .keccak256(ethers.utils.toUtf8Bytes("FirstRandomString"))
+            .substring(0, 34) + ARRAY_INDEX.ZERO,
+          ethers.utils
+            .keccak256(ethers.utils.toUtf8Bytes("FirstRandomString"))
+            .substring(0, 34) + ARRAY_INDEX.ONE,
+          ethers.utils
+            .keccak256(ethers.utils.toUtf8Bytes("FirstRandomString"))
+            .substring(0, 34) + ARRAY_INDEX.TWO,
+          ethers.utils
+            .keccak256(ethers.utils.toUtf8Bytes("FirstRandomString"))
+            .substring(0, 34) + ARRAY_INDEX.THREE,
+        ];
+        const values = [
+          ARRAY_LENGTH.FOUR,
+          ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("FirstRandomString0")
+          ),
+          ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("FirstRandomString1")
+          ),
+          ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("FirstRandomString2")
+          ),
+          ethers.utils.keccak256(
+            ethers.utils.toUtf8Bytes("FirstRandomString3")
+          ),
+        ];
+        const payload = context.token.interface.encodeFunctionData(
+          "setData(bytes32[],bytes[])",
+          [keys, values]
+        );
+
+        await context.keyManager.connect(context.accounts[4]).execute(payload);
+
+        expect(
+          await context.token["getData(bytes32[])"](keys)
+        ).to.be.deep.equal(values);
+      });
+
+      it("should revert when trying to change a key that is not allowed", async () => {
+        const keys = [
+          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("FirstRandomString")),
```

Comment:
not sure to see here which data key is not allowed.

I guess the first one because it does not have 16 x `00`s? Maybe you could hash another word to make this more explicit.

```suggestion
          ethers.utils.keccak256(ethers.utils.toUtf8Bytes("Not Allowed ERC725Y data key")),
```